BedrockServer.cpp

```diff
@@ -920,6 +936,11 @@ void BedrockServer::postPoll(fd_map& fdm, uint64_t& nextActivity) {
             break;
             case STCPManager::Socket::CONNECTED:
             {
+                // We try and read here, so that we're sure to have all the latest data from newly connected sockets.
+                // This is primarily a concern when we're trying to shut down, but had still receiving new requests
+                // right up until the end.
+                s->recv();
```

Comment:
We run this code immediately after accepting new sockets, before doing a `poll` again. It's entirely feasible that those sockets have data waiting on them at this point, and we're trying not to lose requests that we've already received.

It would be easy to wrap this in a check if we're shutting down. If not, we'll be able to wait until the next select loop for data on them, and we'd only do the extra recv() call on shutdown.